fix: maintain agent mutate even when already mutated

a1994sc opened the pull request:
## Description

Removes the checks on the agent webhook for effecting already patched gitops related resources:

- `fluxcd-oci-repo`
- `fluxcd-helm-repo`
- `fluxcd-git-repo`
- `argocd-repository`
- `argocd-application`

The reasoning for removing the this check is that during the regular lifecycle of gitops tools is that they will re-apply the manifests to make sure the cluster reflects the correct state; this means that fluxcd will override the mutated repos to point to something outside of `zarf`'s control, e.i. clear web endpoints that might not be accessible.

## Note

So I did not add tests for removing the check on pods... My logic is that pods are static through their lifecycle once they start, so `zarf` would only need to mutate them once.

## Related Issue

Fixes #3146

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/zarf-dev/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

AustinAbro321 reviewed (changes requested):
Nice work so far! Since this PR is strictly removing code rather than adding code can you delete the test cases involving an already patched Zarf agent, both existing and new

a1994sc commented:
> Nice work so far! Since this PR is strictly removing code rather than adding code can you delete the test cases involving an already patched Zarf agent, both existing and new

So the thought process in why I added the tests for patching even if they already have the labels is so that it leaves some tests for future developers to not remove that functionality... I can remove the tests if you would still like

a1994sc commented:
Thank you Austin, I will update the PR to apply the changes!

a1994sc commented:
@AustinAbro321 Hey, updated the PR to remove the extra tests and added the logic for checking the url against the zarf state registry url. Let me know if that is good

AustinAbro321 reviewed (dismissed):
LGTM, but I'd like to get a review from another one of the @zarf-dev/maintainers before merge

a1994sc commented:
> LGTM, but I'd like to get a review from another one of the @zarf-dev/maintainers before merge

No problem! I figured it was a medium sized change so it is all good to get more eyes on things

phillebaba commented on src/internal/agent/hooks/flux-helmrepo.go:
```diff
@@ -37,6 +37,14 @@ func NewHelmRepositoryMutationHook(ctx context.Context, cluster *cluster.Cluster
+		isPatched bool
+
+		isCreate = r.Operation == v1.Create
+		isUpdate = r.Operation == v1.Update
```
Define these variables where they are first used rather than at the top of the function.

a1994sc replied:
I followed the pattern found in `src/internal/agent/hooks/flux-gitrepo.go`

```
	var (
		patches   []operations.PatchOperation
		isPatched bool

		isCreate = r.Operation == v1.Create
		isUpdate = r.Operation == v1.Update
	)
```
[`ref`](https://github.com/zarf-dev/zarf/blob/a21ed416fc2c2c1c80df97184a49433ee779b933/src/internal/agent/hooks/flux-gitrepo.go#L42-L48) but I can update and get that up tonight

phillebaba commented:
Small change otherwise things look good.

a1994sc commented:
@AustinAbro321 and @phillebaba 

I updated the PR per @phillebaba comments
